### DISCUSSION TITLE
Indexing tool inspiration

### DISCUSSION THREAD

**Discussion by liquescentremedies** [TagStudioDev/TagStudio#155] [Ideas]
I'm only just starting off with development,  but have definitely been making a mess of my file systems for years... One tool I've found which seems to index my files very quickly is Everything. Not sure if that's helpful or not, but it's lightning fast

**Comment by CyanVoxel** [comment]
It looks like Everything takes advantage of USN Journals on NTFS drives, as per their website:
> Does "Everything" monitor file system changes?
> Yes, "Everything" does monitor your file systems for all changes.
> Your search results will update in real-time to reflect any changes.
>
> Everything will automatically keep your NTFS indexes up to date with the NTFS USN Journal.
> Changes will not be missed when Everything is not running as the system maintains the NTFS USN Journal.

While this is specifically for NTFS drives, it would still be helpful in those cases. Another method for fast reading from NTFS drives is by looking that the MFT table, as [WizTree](https://diskanalyzer.com/about) does.
